src/modules/tracker.js

```diff
@@ -2235,6 +2235,390 @@ class Tracker {
+   * @param {number} parameters.numResultsViewed - Number of items viewed in this search result
+   * @param {object[]} [parameters.items] - List of product item objects viewed
+   * @param {string} [parameters.section] - The section name for the item Ex. "Products"
+   * @param {string} [parameters.intentResultId] - Browse result identifier (returned in response from Constructor)
+   * @param {object} [networkParameters] - Parameters relevant to the network request
+   * @param {number} [networkParameters.timeout] - Request timeout (in milliseconds)
+   * @returns {(true|Error)}
+   * @description User viewed a search result within an assistant result
+   * @example
+   * constructorio.tracker.trackAssistantResultView(
+   *     {
+   *         searchResultId: '019927c2-f955-4020-8b8d-6b21b93cb5a2',
+   *         intent: 'show me a recipe for a cookie',
+   *         numResultsViewed: 5,
+   *         items: [{itemId: 'KMH876'}, {itemId: 'KMH140'}, {itemId: 'KMH437'}],
+   *     },
+   * );
+   */
+  trackAssistantResultView(parameters, networkParameters = {}) {
+    // Ensure parameters are provided (required)
+    if (parameters && typeof parameters === 'object' && !Array.isArray(parameters)) {
+      const requestPath = `${this.behavioralV2Url}assistant_search_result_view?`;
+      const {
+        section = 'Products',
+        items,
+        numResultsViewed,
+        intentResultId,
+        searchResultId,
+        intent,
+      } = parameters;
+
+      const bodyParams = {
+        section,
+        intent_result_id: intentResultId,
+        search_result_id: searchResultId,
+        num_results_viewed: numResultsViewed,
+        items: items && Array.isArray(items) && items.slice(0, 100).map((item) => helpers.toSnakeCaseKeys(item, false)),
+        intent,
+      };
+
+      const requestURL = `${requestPath}${applyParamsAsString({}, this.options)}`;
+      const requestMethod = 'POST';
+      const requestBody = applyParams(bodyParams, { ...this.options, requestMethod });
+
+      this.requests.queue(
+        requestURL,
+        requestMethod,
+        requestBody,
+        networkParameters,
+      );
+      this.requests.send();
+
+      return true;
+    }
+
+    this.requests.send();
+
+    return new Error('parameters are required of type object');
+  }
+
+  /**
+   * Send ASA search submitted event
+   *
+   * @function trackAssistantSearchSubmit
+   * @param {object} parameters - Additional parameters to be sent with request
+   * @param {string} parameters.intent - Intent of user request
+   * @param {string} parameters.searchTerm - Term of submitted assistant search event
+   * @param {string} parameters.userInput - The current assistant search query
+   * @param {string} parameters.searchResultId - resultId of search result the clicked item belongs to
+   * @param {string} [parameters.section] - The section name for the item Ex. "Products"
+   * @param {string} [parameters.groupId] - Group identifier of selected item
```

Comment:
Skab document mentions groupId is inside of filter object

Reply:
It says "selected item" but not sure if this is accurate. Should it say selected search term maybe?

Reply:
I'm curious if `groupId` or `filters` would even be applicable in this context. It is part of the original `search_submit` schema, but groupId in the context of search submit doesn't make a whole lot of sense. 🤔